Makefile

```diff
@@ -107,9 +107,11 @@ node-version: node_modules/semver
 
 # ensures that the `node_modules` directory is installed and up-to-date with
 # the dependencies listed in the "package.json" file.
-node_modules: package.json | node-version
+node_modules/*: package.json
```

Comment:
This gets us to allow just running `make run` in the container. Since `node_modules` already exists (due to it being a volume), it would never run. By saying, "the files inside node_modules are dependent on package.json", we guarantee that it will be run inside (and outside) a container.

I tested this pretty rigorously (I've never watched `npm install` happen so many times in my life), but would love this to be better, if it can.